File: JSPM.Web/Modules/Students/StudentsSkills/StudentsSkillsForm.cs
using Serenity.ComponentModel;

namespace JSPM.Students.Forms;

[FormScript("Students.StudentsSkills")]
[BasedOnRow(typeof(StudentsSkillsRow), CheckNames = true)]
public class StudentsSkillsForm
{
    public string SkillCategory { get; set; }
    public int SkillId { get; set; }
    public int Rating { get; set; }
    public string AknowledgeBy { get; set; }
    public string Certification { get; set; }
    public string CertificationDetails { get; set; }
    public string Skills { get; set; }  
    public int StudentId { get; set; }
    
}

File: JSPM.Web/Modules/ServerTypes/Students/StudentsSkillsForm.ts
import { StringEditor, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface StudentsSkillsForm {
    SkillCategory: StringEditor;
    SkillId: IntegerEditor;
    Rating: IntegerEditor;
    AknowledgeBy: StringEditor;
    Certification: StringEditor;
    CertificationDetails: StringEditor;
    StudentId: IntegerEditor;
}

export class StudentsSkillsForm extends PrefixedContext {
    static formKey = 'Students.StudentsSkills';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!StudentsSkillsForm.init)  {
            StudentsSkillsForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;

            initFormType(StudentsSkillsForm, [
                'SkillCategory', w0,
                'SkillId', w1,
                'Rating', w1,
                'AknowledgeBy', w0,
                'Certification', w0,
                'CertificationDetails', w0,
                'StudentId', w1
            ]);
        }
    }
}